backend/src/shortest-path/shortest-path.service.ts

```diff
@@ -0,0 +1,380 @@
+import { Injectable } from '@nestjs/common';
+import { Neo4jService } from 'nest-neo4j/dist';
+import {
+  QueryEdgeResult,
+  QueryResult,
+  ShortestPathQuery,
+} from '../shared/queries';
+import { EdgeDescriptor, NodeDescriptor } from '../shared/entities';
+import { Path, PathEdgeEntry } from './Path';
+import { ShortestPathServiceBase } from './shortest-path.service.base';
+import { FilterService } from '../filter/filter.service';
+import { range } from '../utils';
+import { AppService } from '../app.service';
+
+const KMAP_GDS_GRAPH_NAME_SHORTEST_PATH = 'KMAP_GDS_GRAPH_NAME_SHORTEST_PATH';
+const KMAP_GDS_GRAPH_NAME_SHORTEST_PATH_DIRECTED =
+  'KMAP_GDS_GRAPH_NAME_SHORTEST_PATH_DIRECTED';
+
+@Injectable()
+export class ShortestPathService implements ShortestPathServiceBase {
+  constructor(
+    private readonly neo4jService: Neo4jService,
+    private readonly filterService: FilterService,
+    private readonly queryService: AppService
+  ) {}
+
+  public async executeQuery(query: ShortestPathQuery): Promise<QueryResult> {
+    const path = await this.findShortestPath(
+      query.startNode,
+      query.endNode,
+      query.ignoreEdgeDirections
+    );
+
+    const queryResult = await this.filterService.query(query);
+
+    if (path === null) {
+      return queryResult;
+    }
+
+    // Add the start node to the query result and mark it as path
+    let startNode = queryResult.nodes.find((node) => node.id === path.start.id);
+
+    let startNodeAdded = false;
```

Comment:
Write a comment here, for what purpose this boolean is used, e.g. smth like "this is later used to check if an edgeToAdd is subsidiary"